pkg/controllers/cluster/cluster_controller.go

```diff
@@ -182,6 +183,7 @@ func (c *Controller) Reconcile(ctx context.Context, req controllerruntime.Reques
 	if err := c.Client.Get(ctx, req.NamespacedName, cluster); err != nil {
 		// The resource may no longer exist, in which case we stop processing.
 		if apierrors.IsNotFound(err) {
+			metrics.RemoveClusterStatus(req.NamespacedName.Name)
```

Comment:
The cluster metrics are created by `cluster-status-controller`, It would be better to clean them by the same controller.
To be specifically, I think we can do it at 
https://github.com/karmada-io/karmada/blob/fc1df776dcd3a8ca6a8f5507d90743471ccd066f/pkg/controllers/status/cluster_status_controller.go#L134-L148

Reply:
I update pr and test it!